### PR TITLE
Add remote senior dev for RPC ops fabric

### PR DESCRIPTION
**Pull request by CrashenX**

### CODE REVIEW

File: private_cloud/operational-fabric-developer.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# (Remote) Senior Software Developer - Rackspace Private Cloud
+
+As a Rackspace Private Cloud (RPC) Operational Fabric (ops fabric) developer, you will define, develop, and integrate open source solutions that support our vision: To become the preferred brand for consuming complex open source infrastructure, as-a-Service, delivering industry-leading simplicity, efficiency, and reliability.
```
Comment (jnoller) [inline private_cloud/operational-fabric-developer.md]: I would drop " (ops fabric)" - it's redundant. You may also want to drop the "preferred brand" as that does not speak to the role and is more of an RPC goal, than explaining your team.

File: private_cloud/operational-fabric-developer.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+- **API**: To develop, deploy and support internally facing API(s) for all of the RPC platform and operational fabric allowing for teams/offerings to consume “RPC” as a service in a composable and automated way.
+- **Capacity**: To develop, deploy, and support running services that monitor, track and trend customer resource utilization against available capacity, for better and more consistent utilization management.
+- **Data Collection**: To develop, deploy, and support tools that enable the Analytics team to perform data analysis and insight identification. This includes providing a single access point for customer data including, logging, tickets, run books, and other data sources.
+- **Observability and Controllability**: To develop, deploy, and support services to ensure healthy and continuously running customer environments via modern and standardized operational tooling; ensuring continuity and consistency of service to customers. For example, common monitoring and alerting for operational and support issues and timely resolution (automatically or otherwise).
+
```
Comment (jnoller) [inline private_cloud/operational-fabric-developer.md]: There's a lot here for a candidate to consume; I'd trim aggressively, and explain more about the role itself, what/how/why do they do those things. Which of these are they working on? It's sounds a lot like SRE work, are they going to be on call?

File: private_cloud/operational-fabric-developer.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+The ideal candidates are passionate advocates of technology and open source, and have extensive software design and development experience.
+
+
+### Responsibilities
+
+- Work directly with support engineers and product management to understand what, why, and how to develop and deliver automation.
```
Comment (jnoller) [inline private_cloud/operational-fabric-developer.md]: This feels like an odd ending - "deliver automation" - I think I know what you mean but to an outside reader you may want to rephrase it to be less ambiguous. You also missed operations (support, operations, product management) and users as a stakeholder.

File: private_cloud/operational-fabric-developer.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+- Take high level problems and autonomously deliver solutions supported by data.
+- Architect, design, and develop solutions that leverage and integrate existing open source tools and technologies.
+- Engage with team members primarily via Slack, using Jira/Trello, github, email, and video calls as supplementary communication channels.
+
+### Qualifications
+
+The ideal candidate will have the following (**not** hard requirements):
+
+- BS/MS in Computer Science or equivalent experience.
```
Comment (jnoller) [inline private_cloud/operational-fabric-developer.md]: The trend in the industry has been to omit the BS/MS line as it tends to scare away competent talent with experience.

File: private_cloud/operational-fabric-developer.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+- 5+ years of experience in designing and writing cutting edge software on \*nix platform.
```
Comment (jnoller) [inline private_cloud/operational-fabric-developer.md]: Remove \*nix - is 5 years really required? Also "cutting edge" could also mean cowboy-coding, so I would drop that bit

File: private_cloud/operational-fabric-developer.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+- 3+ years working on highly distributed remote teams.
```
Comment (jnoller) [inline private_cloud/operational-fabric-developer.md]: I'd drop. Simply say "experience working with highly distributed teams"

File: private_cloud/operational-fabric-developer.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+- Worked with several languages and across multiple paradigms.
```
Comment (jnoller) [inline private_cloud/operational-fabric-developer.md]: You're defining a polyglot programmer. I'd scope this in a little more and explain some of the stack they're going to be working with. Also, "paradigms" is business speak and would get lost on the reader.

File: private_cloud/operational-fabric-developer.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+- Experience developing and deploying applications at scale (think thousands of nodes).
```
Comment (jnoller) [inline private_cloud/operational-fabric-developer.md]: developer, deploying and operating - drop (thing thousands of nodes)

File: private_cloud/operational-fabric-developer.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+- Experience with multiple cloud computing technologies.
```
Comment (jnoller) [inline private_cloud/operational-fabric-developer.md]: Such as?

File: private_cloud/operational-fabric-developer.md [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+- Good working knowledge of Internet and operating system security fundamentals.
+- Good working knowledge of message queuing technologies.
+- Good working knowledge of data storage technologies.
```
Comment (jnoller) [inline private_cloud/operational-fabric-developer.md]: Like databases?

File: private_cloud/operational-fabric-developer.md [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+The ideal candidate will have the following (**not** hard requirements):
+
+- BS/MS in Computer Science or equivalent experience.
+- 5+ years of experience in designing and writing cutting edge software on \*nix platform.
+- 3+ years working on highly distributed remote teams.
+- Worked with several languages and across multiple paradigms.
+- Experience developing and deploying applications at scale (think thousands of nodes).
+- Experience with multiple cloud computing technologies.
+- Good working knowledge of Internet and operating system security fundamentals.
```
Comment (jnoller) [inline private_cloud/operational-fabric-developer.md]: Internet? What about network security and secure coding fundamentals

File: private_cloud/operational-fabric-developer.md (hunk 10 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: I was trying to be more general, but more specific is probably better with examples.

File: private_cloud/operational-fabric-developer.md (hunk 11 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: 👍

**Comment by CrashenX** [comment]
Should we add our actual team members and a link to their infos for candidates to see?

File: private_cloud/operational-fabric-developer.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
+# (Remote) Senior Software Developer - Rackspace Private Cloud
+
+As a Rackspace Private Cloud (RPC) Operational Fabric developer, you will define, develop, and integrate open source solutions that support our organizational vision: To become the preferred brand for consuming complex open source infrastructure, as-a-Service, delivering industry-leading simplicity, efficiency, and reliability.
+
+Our team will support this mission by improving observability (i.e. analytics, monitoring, ...) and controllability (i.e. auto-remediation / self-healing, ...) of many clouds, distributed across multiple data-centers.
```
Comment (zack-shoylev) [inline private_cloud/operational-fabric-developer.md]: (i.e. distributed analytics, monitoring, data collection and data processing services)

File: private_cloud/operational-fabric-developer.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+
+### Qualifications
+
+The ideal candidate will have the following (**not** hard requirements):
+
+- Experience architecting and designing production systems that run at scale
+- Experience developing and deploying applications at scale
+- Experience working on highly distributed teams
+- Built production software in multiple languages (e.g. Go, JS, Haskell, Python, Scala, ...)
+- Worked across multiple programming paradigms (declaritive, imperative, functional, OO)
```
Comment (zack-shoylev) [inline private_cloud/operational-fabric-developer.md]: I would trim this or combine it with the line above.

File: private_cloud/operational-fabric-developer.md [hunk 14]
```diff
@@ -0,0 +1,44 @@
+# (Remote) Senior Software Developer - Rackspace Private Cloud
+
+As a Rackspace Private Cloud (RPC) Operational Fabric developer, you will define, develop, and integrate open source solutions that support our organizational vision: To become the preferred brand for consuming complex open source infrastructure, as-a-Service, delivering industry-leading simplicity, efficiency, and reliability.
```
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: organizational vision, which is to deliver industry-leading products that are reliable, efficient, and simple to use. Our aim is to become the preferred choice for developers who want to consume open-source infrastructure as a service.

File: private_cloud/operational-fabric-developer.md (hunk 12 above)
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: Our team supports this mission by improving how we observe and ensure the reliable running of multiple clouds distributed across multiple data centers through monitoring, auto-remediation and self-healing,

File: private_cloud/operational-fabric-developer.md [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+
+Our team will support this mission by improving observability (i.e. analytics, monitoring, ...) and controllability (i.e. auto-remediation / self-healing, ...) of many clouds, distributed across multiple data-centers.
+
+The ideal candidates are passionate advocates of technology and open source, and have extensive software design and development experience.
```
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: Not sure "technology" adds anything here, can we remove?

File: private_cloud/operational-fabric-developer.md [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+
+### About Us
+
+- **Team Size**: Currently 4 senior devs (looking to add 2-3 more) and a technical manager
+- **Hours**: We have flexible working hours and aim for a median of 35-40 hours a week
```
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: 👍 Great to point this out

File: private_cloud/operational-fabric-developer.md [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+- **On-Call**: No, though deploying new systems will require a level of care and feeding
+- **Travel**: About once or twice a year for a conference or team gathering
```
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: Might also be worth pointing out this a remote position

File: private_cloud/operational-fabric-developer.md [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+
+### Responsibilities
+
+- Work directly with support engineers, operations, product management, and end users
```
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: end-users

File: private_cloud/operational-fabric-developer.md [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+- Write automation to improve efficiency, reliability, and simplicity of clouds
```
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: "Use automation" or "Write automation scripts"

File: private_cloud/operational-fabric-developer.md [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+- Take high level problems and autonomously deliver solutions supported by data
```
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: How about "data-driven solutions"?

File: private_cloud/operational-fabric-developer.md [hunk 21]
```diff
@@ -0,0 +1,44 @@
...
+- Architect, design, and develop solutions that leverage and integrate existing open source tools and technologies
```
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: Can we remove "architect", since I don't think there's much of a difference with "design"

File: private_cloud/operational-fabric-developer.md [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
+- Engage asynchronously with team members via Slack, Jira/Trello, github, and email
+- Engage synchronously with team members via video calls
+
+### Qualifications
+
+The ideal candidate will have the following (**not** hard requirements):
+
+- Experience architecting and designing production systems that run at scale
```
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: Again, do we need architect if we have design? Also, what about operating?

File: private_cloud/operational-fabric-developer.md [hunk 23]
```diff
@@ -0,0 +1,44 @@
...
+- Experience developing and deploying applications at scale
+- Experience working on highly distributed teams
```
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: Do we need "highly"?

File: private_cloud/operational-fabric-developer.md (hunk 13 above)
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: declarative
object-orientated (you didn't define this acronym already)

File: private_cloud/operational-fabric-developer.md [hunk 24]
```diff
@@ -0,0 +1,44 @@
...
+- Built production software in multiple languages (e.g. Go, JS, Haskell, Python, Scala, ...)
```
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: How about "Familiarity with different languages (e.g. ...)" You're also missing Ruby and PHP - is that intentional?

File: private_cloud/operational-fabric-developer.md [hunk 25]
```diff
@@ -0,0 +1,44 @@
...
+- Worked across multiple programming paradigms (declaritive, imperative, functional, OO)
+- Experience with multiple cloud computing technologies (e.g. OpenStack, K8s)
```
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: AWS, GCE? Also, I wouldn't say k8s is necessarily cloud.

File: private_cloud/operational-fabric-developer.md [hunk 26]
```diff
@@ -0,0 +1,44 @@
...
+- Good working knowledge of network security and secure coding fundamentals
```
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: What's a secure coding fundamental? TLS? Encryption? Can we be specific

Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: I think above all "garbage in" / "garbage out" (i.e. sanitize inputs). Also, understanding trade-offs for CIA (Confidentiality, Integrity, and Availability). Things like that. Not sure what I would put as an i.e. or e.g. there.

File: private_cloud/operational-fabric-developer.md (hunk 25 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: 👍

File: private_cloud/operational-fabric-developer.md (hunk 13 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: 👍

File: private_cloud/operational-fabric-developer.md (hunk 24 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: Not an exhaustive list, not intentional. That's why I used e.g. here.

File: private_cloud/operational-fabric-developer.md (hunk 23 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: Probably not, 👍

File: private_cloud/operational-fabric-developer.md (hunk 22 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: Yes to the former, 👍  to the latter

Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: Although I think the latter goes in the next bullet

File: private_cloud/operational-fabric-developer.md (hunk 21 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: I fundamental do, and I think it's important specifically for this role.

File: private_cloud/operational-fabric-developer.md (hunk 20 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: k, will try to reword

File: private_cloud/operational-fabric-developer.md (hunk 19 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: Either write or develop. Also, I don't like the word script. It diminishes what's being done here and programming work in general.

File: private_cloud/operational-fabric-developer.md (hunk 18 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: 👍

File: private_cloud/operational-fabric-developer.md (hunk 17 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: Yeah, maybe I'll but a location bullet

File: private_cloud/operational-fabric-developer.md (hunk 15 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: k

File: private_cloud/operational-fabric-developer.md (hunk 14 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: Those words are taken directly from Crenshaw's all hands deck.

File: private_cloud/operational-fabric-developer.md (hunk 13 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: Yeah, I had it combined and it was kinda long so, I split apart, heh

File: private_cloud/operational-fabric-developer.md (hunk 12 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: 👍

File: private_cloud/operational-fabric-developer.md (hunk 14 above)
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: Fair enough ... 🤐

File: private_cloud/operational-fabric-developer.md (hunk 19 above)
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: Yeah, that's a fair point. How about "Write automation software"?

File: private_cloud/operational-fabric-developer.md (hunk 21 above)
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: Fair enough 👍

File: private_cloud/operational-fabric-developer.md (hunk 24 above)
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: Yeah, I guess so. I was just a bit caught off guard when I saw Haskell put before Python, since order might indicate preference to some folks.

File: private_cloud/operational-fabric-developer.md (hunk 26 above)
Comment (jamiehannaford) [inline private_cloud/operational-fabric-developer.md]: how about

> and secure coding fundamentals, such as TLS, how to protect against common attack vectors (e.g. using input validation) and the trade-offs for CIA (Confidentiality, Integrity, and Availability).

File: private_cloud/operational-fabric-developer.md (hunk 20 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: actually, I like what I'm trying to convey here as is

File: private_cloud/operational-fabric-developer.md (hunk 22 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: Actually, in this case. I'm primarily interested in architecting. Reworded.

File: private_cloud/operational-fabric-developer.md (hunk 24 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: I think familiarity is acceptable, but prefer built production software in

File: private_cloud/operational-fabric-developer.md (hunk 26 above)
Comment (CrashenX) [inline private_cloud/operational-fabric-developer.md]: Added SSL and GIGO

File: private_cloud/operational-fabric-developer.md [hunk 27]
```diff
@@ -0,0 +1,51 @@
...
+- Built production software in multiple languages (e.g. Go, Haskell, JS, Python, Ruby, Scala)
+- Worked across multiple programming paradigms (i.e. declarative, functional, imperative, object-oriented)
+- Experience with multiple cloud computing technologies (e.g. AWS, GCE, OpenStack)
+- Good working knowledge of network security and secure coding fundamentals (e.g. SSL, GIGO)
+- Good working knowledge of messaging technologies (e.g. RabbitMQ, Kafka)
+- Good working knowledge of both document and relational databases (e.g. Mongo, MySQL)
+- Good working knowledge of monitoring technologies (e.g. Kapacitor, Telegraf)
+- Good working knowledge of analytics technologies (e.g. R, Pandas)
+- Excellent written and verbal skills
+
```
Comment (supermari0) [inline private_cloud/operational-fabric-developer.md]: Maybe explicitly calling out something around testing would be good? Otherwise LGTM 👍
